Code/Tools/ProjectManager/Source/Application.cpp

```diff
@@ -24,6 +27,73 @@
 #include <QInputDialog>
 #include <QIcon>
 
+#if defined(AZ_PLATFORM_WINDOWS)
+LONG WINAPI CreateMiniDump(struct _EXCEPTION_POINTERS* exceptionPointers)
+{
+    AZ::IO::FixedMaxPath dumpPath{ "@log@/O3DEDump.dmp" };
+    if (auto fileIoBase = AZ::IO::FileIOBase::GetInstance(); fileIoBase != nullptr)
+    {
+        dumpPath = fileIoBase->ResolvePath(dumpPath, "@log@/O3DEDump.dmp");
+    }
+
+    // FixedMaxPath appends '\x1' to the end of the path string which causes CreateFileW to fail, remove it
+    AZStd::string dumpPathStr = dumpPath.String();
+    dumpPathStr = dumpPathStr.substr(0, dumpPathStr.length() - 1);
+    const char* dumpPathRaw = dumpPathStr.c_str();
+
+    MINIDUMP_TYPE dumpType = MINIDUMP_TYPE::MiniDumpNormal;
+
+    fflush(nullptr);
+    HMODULE hndDBGHelpDLL = LoadLibraryA("DBGHELP.DLL");
+
+    if (!hndDBGHelpDLL)
+    {
+        AZ_Warning("ProjectManager", false, "Failed to record DMP file: Could not open DBGHELP.DLL");
```

Comment:
I'm wondering if all these AZ_Warnings should be MessageBoxes instead if it's possible the AZ_Warning will never make it into the logs.

Reply:
While the logger is initialized, very early in the program then this will show up. So if the failure to report occurs early in the application startup itself then they might not show. IMO I think it's fine's fine. More useful seeing this info in the log.

Reply:
I'm OK with this for now, but when crashing, all bets in your own code are often off - especially in the case of stack/heap corruption.  The crash could in fact be occurring inside the AZ debug logging system used by AZ_Warning.